docs/Setting-up-monorepo.md

```diff
@@ -0,0 +1,357 @@
+
+1. Install Ignite CLI (if you haven't already):
+
+```shell
+npx ignite-cli@latest
+```
+
+2. Generate a new app:
+   Navigate to the apps/ directory and run the following command to create a new app:
+
+```shell
+cd apps
+ignite new mobile
+```
+
+We suggest the following answers to the prompts:
+
+```
+📝 Do you want to use Expo?: Expo - Recommended for almost all apps [Default]
+📝 Which Expo workflow?: Expo Go - For simple apps that don't need custom native code [Default]
+📝 Do you want to initialize a git repository?: No
+📝 Remove demo code? We recommend leaving it in if it's your first time using Ignite: No
+📝 Which package manager do you want to use?: yarn
+📝 Do you want to install dependencies?: No
+```
+
+3. Open `metro.config.js` file
+
+```shell
+touch mobile/metro.config.js
+```
+
+4. Replace Metro configuration with the following content
+
+```
+const { getDefaultConfig } = require('expo/metro-config');
+const path = require('path');
+
+// Find the project and workspace directories
+const projectRoot = __dirname;
+// This can be replaced with `find-yarn-workspace-root`
+const monorepoRoot = path.resolve(projectRoot, '../..');
+
+const config = getDefaultConfig(projectRoot);
+
+// 1. Watch all files within the monorepo
+config.watchFolders = [monorepoRoot];
+// 2. Let Metro know where to resolve packages and in what order
+config.resolver.nodeModulesPaths = [
+  path.resolve(projectRoot, 'node_modules'),
+  path.resolve(monorepoRoot, 'node_modules'),
+];
+
+module.exports = config;
+```
+
+## Step 3: Create web app
+
+Let's create a second app now. This app is meant to be run on web browsers.
+
+1. Initialize web app
+
+```shell
+cd ..
+mkdir web
+cd web
+yarn init
+```
+
+2. Create an `index.html` file with a simple form:
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Simple Form App</title>
+</head>
+<body>
+  <h1>Sign Up</h1>
+  <form id="signup-form">
+    <div>
+      <label for="email">Email:</label>
+      <input type="email" id="email" name="email" required>
+    </div>
+    <div>
+      <label for="password">Password:</label>
+      <input type="password" id="password" name="password" required>
+    </div>
+    <button type="submit">Submit</button>
+  </form>
+
+  <script src="index.js" type="module"></script>
+</body>
+</html>
+```
+
+3. Add a simple development server
+   To serve your app with a local server, install a simple development server like `live-server`
+
+```shell
+yarn add live-server -D
+```
+
+4. Add a script to your `package.json` to run the server:
+
+```json
+"scripts": {
+  "start": "live-server"
+}
+```
+
+## Step 4: Install dependencies
+
+Let's make sure all of our dependendencies are installed for both apps.
+
+1. Run `yarn` at the root of the project
+
+```shell
+cd ..
+cd ..
+yarn
+```
+
```

Comment:
It might be good to talk a bit about what this did -- i.e. point out where the `node_modules` folders are, with a link to something about hoisting.

Also, you may want a note for users of yarn 2+ about checking their settings in `yarnrc.yml` in case things are hoisted that shouldn't be. Not like full instructions just a pointer to the issue.

Reply:
Hey @trevor-coleman, thanks! 

Yes, did add a section for your first point. 

For the second one, this guide is intended for Yarn 3.x. Do you think it would still be valid?

Reply:
Ah no if it's yarn 3 then you wouldn't need it.